src/dune_rules/lib.ml

```diff
@@ -832,23 +830,17 @@ end = struct
 
   let empty = { stack = []; seen = Id.Set.empty; implements_via = Id.Map.empty }
 
-  let to_required_by t ~stop_at =
-    let stop_at = stop_at.stack in
+  let to_required_by t =
     let rec loop acc l =
-      if List.physically_equal l stop_at then
-        List.rev acc
-      else
-        match l with
-        | [] -> assert false
-        | ({ Id.path; name; _ } as id) :: l ->
-          let implements_via =
-            let open Option.O in
-            let+ via = Id.Map.find t.implements_via id in
-            Implements_via.to_dep_path_implements_via via
-          in
-          loop
-            ({ Dep_path.Entry.lib = { path; name }; implements_via } :: acc)
-            l
+      match l with
+      | [] -> assert false
```

Comment:
@jeremiedimino did you forget to edit this base case? I can't see how this function can ever work with this `assert false`

Reply:
Indeed. In fact it's just a `List.map` now. I updated the code.